pkg/inventory/model/client.go

```diff
@@ -351,11 +334,16 @@ func (r *Client) Watch(model Model, handler EventHandler) (w *Watch, err error)
 			w = nil
 		}
 	}()
-	itr, err := Table{r.db}.Iter(
-		model,
-		ListOptions{Detail: 1})
+	var itr fb.Iterator
+	func() {
+		r.dbMutex.RLock()
+		defer r.dbMutex.RUnlock()
+		itr, err = Table{r.db}.Iter(
+			model,
+			ListOptions{Detail: 1})
+	}()
```

Comment:
Wrap the _list_ operation in a function that acquires the read lock and defer the unlock.